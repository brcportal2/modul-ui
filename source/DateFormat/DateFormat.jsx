import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {dateHelper} from 'modul-helpers';

export default class DateFormat extends Component {
	static propTypes = {
		value: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.instanceOf(Date)
		]),
		format: PropTypes.string,
		def: PropTypes.string,
		className: PropTypes.string,
	};

	static defaultProps = {
		format: 'clever',
		def: '',
		className: ''
	};

	render() {
		const {
			value,
			format,
			def,
			className
		} = this.props;

		let formatDate = value;

		if (value && typeof value === 'string') {
			formatDate = dateHelper.stringToDate(value);

			if (isNaN(formatDate.getTime()))
				formatDate = new Date(value);
		}

		if (formatDate && formatDate instanceof Date)
			return (
				<span className={className}>
					{dateHelper.dateFormat(formatDate, format)}
				</span>
			);

		return (
			<span>
				{def}
			</span>
		);
	}
}