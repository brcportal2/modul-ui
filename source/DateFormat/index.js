import DateFormat from './DateFormat';


export default DateFormat;
export {
    DateFormat
};