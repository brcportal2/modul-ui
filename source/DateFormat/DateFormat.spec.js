import React from 'react';
import renderer from 'react-test-renderer';

import DateFormat from './DateFormat';


describe('DateFormat', () => {
    test('Default props', () => {
        const component = renderer.create(
            <DateFormat
                value='12.02.2012'
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Default value', () => {
        const component = renderer.create(
            <DateFormat
                def='not date'
                value='12.02.2012'
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('ClassName', () => {
        const component = renderer.create(
            <DateFormat
                className='test_class'
                def='not date'
                value='12.02.2012'
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Null value', () => {
        const component = renderer.create(
            <DateFormat
                className='test_class'
                def='not date'
                value={null}
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Masks value', () => {
        const component = renderer.create(
            <DateFormat
                className='test_class'
                def='not date'
                format='cleverDate'
                value='12.02.2012 12:15'
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Date value', () => {
        const component = renderer.create(
            <DateFormat
                className='test_class'
                def='not date'
                format='cleverDate'
                value={new Date(999999999999)}
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});