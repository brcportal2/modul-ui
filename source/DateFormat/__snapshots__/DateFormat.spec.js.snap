// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateFormat ClassName 1`] = `
<span
  className="test_class"
>
  02.12.2012 00:00
</span>
`;

exports[`DateFormat Date value 1`] = `
<span
  className="test_class"
>
  09.09.2001
</span>
`;

exports[`DateFormat Default props 1`] = `
<span
  className=""
>
  02.12.2012 00:00
</span>
`;

exports[`DateFormat Default value 1`] = `
<span
  className=""
>
  02.12.2012 00:00
</span>
`;

exports[`DateFormat Masks value 1`] = `
<span
  className="test_class"
>
  02.12.2012
</span>
`;

exports[`DateFormat Null value 1`] = `
<span>
  not date
</span>
`;
